lib/phonebook/ui/components/kinds_bar.dart

```diff
@@ -1,41 +1,59 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
 import 'package:hooks_riverpod/hooks_riverpod.dart';
 import 'package:myecl/phonebook/providers/association_kind_provider.dart';
 import 'package:myecl/phonebook/providers/association_kinds_provider.dart';
 import 'package:myecl/tools/ui/builders/async_child.dart';
-import 'package:myecl/tools/ui/layouts/horizontal_list_view.dart';
 import 'package:myecl/tools/ui/layouts/item_chip.dart';
 
-class KindsBar extends ConsumerWidget {
-  const KindsBar({super.key});
-
+class KindsBar extends HookConsumerWidget {
+  KindsBar({super.key});
+  final dataKey = GlobalKey();
   @override
   Widget build(BuildContext context, WidgetRef ref) {
     final kind = ref.watch(associationKindProvider);
     final kindNotifier = ref.watch(associationKindProvider.notifier);
     final associationKinds = ref.watch(associationKindsProvider);
-
+    useEffect(
```

Comment:
Can you find a way to not use useEffect, maybe by using a scrollcontroller

Reply:
Why is that a problem ? It actually does what I want by applying the effect when the key is rebuilt, on page build and on chip selection